presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -135,14 +136,16 @@ public OrderingScheme map(OrderingScheme orderingScheme)
     {
         // SymbolMapper inlines symbol with multiple level reference (SymbolInliner only inline single level).
         ImmutableList.Builder<VariableReferenceExpression> orderBy = ImmutableList.builder();
-        ImmutableMap.Builder<VariableReferenceExpression, SortOrder> ordering = ImmutableMap.builder();
+        HashMap<VariableReferenceExpression, SortOrder> orderingMap = new HashMap<VariableReferenceExpression, SortOrder>();
         for (VariableReferenceExpression variable : orderingScheme.getOrderByVariables()) {
             VariableReferenceExpression translated = map(variable);
-            orderBy.add(translated);
-            ordering.put(translated, orderingScheme.getOrdering(variable));
+            // Some variables may become duplicates after canonicalization, so we put them only once.
+            if (!orderingMap.containsKey(translated)) {
```

Comment:
When you are at it, maybe warn if someone does 

ORDER BY x DESC, x ASC

because currently we just take the first one. But since you are checking here, you can warn.

Reply:
Should that be a separate PR?

Reply:
Actually let's do it in this PR as well, cause otherwise by deduping we might hide the error.

Reply:
Actually, I say give a warning anyway if you see a repetition and also a second warning if they are mismatched sort direction.

Reply:
I checked this information is already lost when OrderingScheme is created. We can only add a warning here if we have 

ORDER BY x DESC, y ASC

where x and y are same canonical variables, but different variables in input.

To put a warning for all cases of same variable, we need to have a different PR which touches different points where OrderingScheme is created

Reply:
OK. Create an issue so we can track it.

Reply:
https://github.com/prestodb/presto/issues/16327